recipes/Plan_Solve/Plan_Solve_Agent.ipynb

```diff
@@ -0,0 +1,396 @@
+    "- **watsonx.ai Access**: You'll need credentials for IBM's watsonx.ai platform\n",
+    "- **API Keys**: Optional API keys for weather and stock market data (with fallback demo data)\n",
+    "\n",
+    "### Required Credentials\n",
+    "You'll need to set up the following environment variables:\n",
+    "- `WATSONX_URL`: Your watsonx.ai endpoint URL\n",
+    "- `WATSONX_APIKEY`: Your watsonx.ai API key  \n",
+    "- `WATSONX_PROJECT_ID`: Your watsonx.ai project ID\n",
+    "- `WEATHER_API_KEY`: (Optional) OpenWeatherMap API key for real weather data\n",
+    "- `AV_STOCK_API_KEY`: (Optional) Alpha Vantage API key for real stock data\n",
+    "\n",
+    "\n",
+    "You can run this notebook in [Colab](https://colab.research.google.com/), or download it to your system and [run the notebook locally](https://github.com/ibm-granite-community/granite-kitchen/blob/main/recipes/Getting_Started_with_Jupyter_Locally/Getting_Started_with_Jupyter_Locally.md). \n",
+    "\n",
+    "To avoid Python package dependency conflicts, we recommend setting up a [virtual environment](https://docs.python.org/3/library/venv.html).\n",
+    "\n",
+    "Note, this notebook is compatible with Python 3.12 and well as Python 3.11, the default in Colab at the time of publishing this recipe. To check your python version, you can run the `!python --version` command in a code cell.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Check Python version\n",
+    "!python --version"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! echo \"::group::Install Dependencies\"\n",
+    "%pip install uv\n",
+    "! uv pip install \"git+https://github.com/ibm-granite-community/utils.git\" \\\n",
+    "    langgraph \\\n",
+    "    langchain \\\n",
+    "    langchain_ibm\n",
+    "! echo \"::endgroup::\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5",
+   "metadata": {},
+   "source": [
+    "If you run this notebook in [Colab](https://colab.research.google.com/), create a `Plan_Solve` folder on your google drive with all the content from this repository including .env file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dotenv import load_dotenv\n",
+    "from IPython.display import Image, display\n",
+    "try:\n",
+    "    import sys\n",
+    "    import google.colab\n",
+    "    from google.colab import drive\n",
+    "    drive.mount('/content/drive')\n",
+    "    sys.path.append('/content/drive/My Drive/Plan_Solve')\n",
+    "    load_dotenv('/content/drive/My Drive/Plan_Solve/.env')\n",
+    "    print('🌐 Notebook Running Locally ')\n",
+    "except:\n",
+    "    print('💻 Notebook Running Locally ')\n",
+    "    load_dotenv()"
```

Comment:
The notebook should be monolithic and self contained. So one could run it in colab. The code here assume your personal Google Drive rather than the contents of the git repo at the revision of the notebook.